public/js/actions/sources.js

```diff
@@ -102,14 +105,14 @@ function selectSource(id, options = {}) {
       return;
     }
 
-    const source = getSource(getState(), id).toJS();
+    const source = getSource(getState(), id);
```

Comment:
The contract for actions is that they always receive normal JS objects. This gives us some sanity about the API, because components can directly fire actions, and they should always know that that can simply pass literal objects like `{ sourceId: ... }`. So please move `toJS` back here, even if right now that action creator is only fired internally.